src/construction.cpp

```diff
@@ -472,8 +485,13 @@ void construction_menu()
 
         const std::string action = ctxt.handle_input();
         const long raw_input_char = ctxt.get_raw_input().get_first_input();
-
-        if( action == "DOWN" ) {
+        if( action == "FILTER" ){
+            update_info = true;
+            update_cat = true;
+            tabindex = 9;
```

Comment:
`uimenu` does that for all entries. Though it would probably require redesigning the menu to be a `uimenu`, which would be out of scope of this PR.

It's fine for it to act like crafting menu, which also seems to have it in `string_input_popup` variant.